pkg/services/ngalert/remote/alertmanager.go

```diff
@@ -194,6 +195,46 @@ func (am *Alertmanager) CompareAndSendConfiguration(ctx context.Context, config
 	return nil
 }
 
+func (am *Alertmanager) SendConfiguration(ctx context.Context, config *models.AlertConfiguration) error {
```

Comment:
This is the wrong place - you need to decrypt and then compare, otherwise if someone changes a secret of their configuration it won't be sent.

Reply:
Good catch! It would always be sent though, hashes would always be different, so fixing this would save us unnecessary sends.